tests/Oryx.Tests.Common/EndToEndTestHelper.cs

```diff
@@ -204,11 +216,34 @@ public static class EndToEndTestHelper
 
                         break;
                     }
-                    catch (Exception ex) when (ex.InnerException is IOException || ex.InnerException is SocketException)
+                    catch (Exception ex) when (ex.InnerException is IOException ||
+                    ex.InnerException is SocketException)
                     {
                         if (i == MaxRetryCount - 1)
                         {
-                            output.WriteLine(runResult.GetDebugInfo());
+                            string debugInfo = string.Empty;
+
+                            // ToString() on StringBuilder is throwing an exception probably because of a 
+                            // multithreading issue where the container is still writing data into it and we are trying
+                            // to retrieve the content out of it.
+                            try
+                            {
+                                // TO i
+                                // System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative 
```

Comment:
cc @arroyc @gpcastro I ran into the following issue again with my PR
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2656104

On searching the error message, seems like it could be a multi-threading issue with stringbuilder